mm2src/coins/eth/eth_swap_v2/eth_taker_swap_v2.rs

```diff
@@ -1,31 +1,36 @@
-use super::eth::{wei_from_big_decimal, EthCoin, EthCoinType, SignedEthTx, TAKER_SWAP_V2};
-use super::{decode_contract_call, get_function_input_data, ParseCoinAssocTypes, RefundFundingSecretArgs,
-            RefundTakerPaymentArgs, SendTakerFundingArgs, SwapTxTypeWithSecretHash, TakerPaymentStateV2, Transaction,
-            TransactionErr, ValidateSwapV2TxError, ValidateSwapV2TxResult, ValidateTakerFundingArgs};
+use super::{check_decoded_length, validate_from_to_and_status, validate_payment_args, validate_payment_state,
+            EthPaymentType, PaymentMethod, PrepareTxDataError, ZERO_VALUE};
+use crate::eth::{decode_contract_call, get_function_input_data, wei_from_big_decimal, EthCoin, EthCoinType,
+                 ParseCoinAssocTypes, RefundFundingSecretArgs, RefundTakerPaymentArgs, SendTakerFundingArgs,
+                 SignedEthTx, SwapTxTypeWithSecretHash, TakerPaymentStateV2, TransactionErr, ValidateSwapV2TxError,
+                 ValidateSwapV2TxResult, ValidateTakerFundingArgs, TAKER_SWAP_V2};
 use crate::{FundingTxSpend, GenTakerFundingSpendArgs, GenTakerPaymentSpendArgs, SearchForFundingSpendErr,
-            WaitForTakerPaymentSpendError};
+            WaitForPaymentSpendError};
 use common::executor::Timer;
 use common::now_sec;
-use enum_derives::EnumFromStringify;
-use ethabi::{Contract, Function, Token};
+use ethabi::{Function, Token};
 use ethcore_transaction::Action;
 use ethereum_types::{Address, Public, U256};
 use ethkey::public_to_address;
 use futures::compat::Future01CompatExt;
 use mm2_err_handle::prelude::{MapToMmResult, MmError, MmResult};
-use mm2_number::BigDecimal;
 use std::convert::TryInto;
-use web3::types::{Transaction as Web3Tx, TransactionId};
+use web3::types::TransactionId;
 
-/// ZERO_VALUE is used to represent a 0 amount in transactions where the value is encoded in the transaction input data.
-/// This is typically used in function calls where the value is not directly transferred with the transaction, such as in
-/// `spendTakerPayment` where the [amount](https://github.com/KomodoPlatform/etomic-swap/blob/5e15641cbf41766cd5b37b4d71842c270773f788/contracts/EtomicSwapTakerV2.sol#L166)
-/// is provided as part of the input data rather than as an Ether value
-pub(crate) const ZERO_VALUE: u32 = 0;
 const ETH_TAKER_PAYMENT: &str = "ethTakerPayment";
 const ERC20_TAKER_PAYMENT: &str = "erc20TakerPayment";
 const TAKER_PAYMENT_APPROVE: &str = "takerPaymentApprove";
 
+/// state index for `TakerPayment` structure from `EtomicSwapTakerV2.sol`
+///
+///     struct TakerPayment {
+///         bytes20 paymentHash;
+///         uint32 preApproveLockTime;
+///         uint32 paymentLockTime;
+///         TakerPaymentState state;
+///     }
+const TAKER_PAYMENT_STATE_INDX: usize = 3;
```

Comment:
`TAKER_PAYMENT_STATE_INDEX`?

Reply:
didnt include E intentionally (dont like long names), but as it causes questions, I updated const names https://github.com/KomodoPlatform/komodo-defi-framework/pull/2211/commits/ed023922aa5ada0d4ff592279e2a4bad1453e514